OpenSim/Sandbox/futureMuscleMetabolicsResponse.cpp

```diff
@@ -0,0 +1,210 @@
+public:
+    OpenSim_DECLARE_PROPERTY(strength, double, "per-coord param.");
+    ComplexResponse() {
+        constructInfrastructure();
+        constructProperty_strength(3.0);
+    }
+    double getTerm1(const State& s) const {
+        if (!isCacheVariableValid(s, "term_1")) {
+            const auto& coord = getConnectee<Coordinate>("coord");
+            const auto value = coord.getValue(s);
+            setCacheVariableValue(s, "term_1", get_strength() * value);
+            markCacheVariableValid(s, "term_1");
+        }
+        return getCacheVariableValue<double>(s, "term_1");
+    }
+    double getTerm2(const State& s) const {
+        if (!isCacheVariableValid(s, "term_2")) {
+            const auto& coord = getConnectee<Coordinate>("coord");
+            const auto speed = coord.getSpeedValue(s);
+            setCacheVariableValue(s, "term_2", 2.0 * speed);
+            markCacheVariableValid(s, "term_2");
+        }
+        return getCacheVariableValue<double>(s, "term_2");
+    }
+    double getTotal(const State& s) const {
+        if (!isCacheVariableValid(s, "sum")) {
+            const auto& coord = getConnectee<Coordinate>("coord");
+            const auto speed = coord.getSpeedValue(s);
+            setCacheVariableValue(s, "sum", getTerm1(s) + getTerm2(s));
+            markCacheVariableValid(s, "sum");
+        }
+        return getCacheVariableValue<double>(s, "sum");
+    }
+private:
+    void constructConnectors() override {
+        constructConnector<Coordinate>("coord");
+    }
+    void constructOutputs() override {
+        constructOutput<double>("term_1",
+                &ComplexResponse::getTerm1, SimTK::Stage::Position);
+        constructOutput<double>("term_2",
+                &ComplexResponse::getTerm2, SimTK::Stage::Velocity);
+        constructOutput<double>("sum", &ComplexResponse::getTotal,
+                SimTK::Stage::Velocity);
+    }
+    void extendAddToSystem(MultibodySystem& system) const override {
+        Super::extendAddToSystem(system);
+        addCacheVariable<double>("term_1",
+                0.0, SimTK::Stage::Velocity);
+        addCacheVariable<double>("term_2",
+                0.0, SimTK::Stage::Velocity);
+        addCacheVariable<double>("sum",
+                0.0, SimTK::Stage::Velocity);
+    }
+};
+
+class AggregateResponse : public ModelComponent {
+    OpenSim_DECLARE_CONCRETE_OBJECT(AggregateResponse, ModelComponent);
+public:
+    AggregateResponse() { constructInfrastructure(); }
+// TODO want to use list property, but that makes clones and inputs/outputs
+// don't get copied yet.
+//    OpenSim_DECLARE_LIST_PROPERTY(responses, ComplexResponse,
+//            "for individual coordinates.");
+    // Temporary solution:
+    std::vector<std::shared_ptr<ComplexResponse>> responses;
+
+    // TODO how do we propagate this scaling_factor to the ComplexResponses?
+    OpenSim_DECLARE_PROPERTY(scaling_factor, double, "Affects each coord.");
```

Comment:
Now (as of #634) that each component can have a parent, you can have the child component check if it has a parent, and if so to get its scaling_factor.